### PR TITLE
Add blog post about deprecating nodes

### PR DESCRIPTION
**Pull request by daviddavis**
ref #2460
https://pulp.plan.io/issues/2460

### CODE REVIEW

File: _posts/2016-12-06-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-06-deprecating-nodes.md]: s/in sync/synchronized/     maybe?

File: _posts/2016-12-06-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-06-deprecating-nodes.md]: Nodes needs to be deprecated with 2.12 since 2.11 is already at beta and has been branched. This means that 2.11-dev is already 2.11.1.

File: _posts/2016-12-06-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-06-deprecating-nodes.md]: I think there is more to be said than the maintenance effort. What about this for the first paragraph which includes the above comments:

Nodes were introduced in Pulp in 2.1.0 and provided a way to keep multiple Pulp servers synchronized. However, Nodes uses separate codepaths from the normal sync+publish codepaths which creates at a few problems:

* Nodes is less tested than sync+publish even though it provides a similar feature set. This greatly reduces the user to user usage and testing benefit.
* Fixing bugs in sync+publish does not fix bugs in Nodes and vice-versa
* Nodes has a good amount of bugs already reported that are still unfixed

Given that much of the functionality of Nodes can be achieved by running Pulp with sync+publish workflow, the core team is removing Nodes in the next major release of Pulp 3.0. With Pulp 2.12, Nodes will be officially deprecated.

File: _posts/2016-12-06-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-06-deprecating-nodes.md]: s/configured the same way/each a standard Pulp server installation without any Nodes packages installed/.

File: _posts/2016-12-06-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-06-deprecating-nodes.md]: s/simply/called that because it is/

File: _posts/2016-12-06-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-06-deprecating-nodes.md]: s/First/First,/

File: _posts/2016-12-06-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-06-deprecating-nodes.md]: s/simply//

File: _posts/2016-12-06-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-06-deprecating-nodes.md]: s/centos-base/'centos-base'/
s/my/the/

File: _posts/2016-12-06-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-06-deprecating-nodes.md]: What about adding an example to cause cron to repeat publish from the parent?

Also one gotcha with this approach is that pulp-admin cannot take the pulp server url to talk to as a parameter (iirc). We could recommend that the parent server and child both have pulp-admin configured to talk to localhost and that the parent jobs live in cron on the parent and the child jobs on the child. I'm not sure how to incorporate this content, but it's something I wanted to mention.

**Comment by bmbouter** [comment]
FYI, this repo is not monitored by pulp.plan.io so you'll have to manually move the issue to MODIFIED after it is merged and you won't be able to associate the issue. That is all normal (for blog posts) but I wanted to relay the info.

**Review by bmbouter — CHANGES_REQUESTED** [review]
Thank you for putting this together! I really like it. I left some comments requested. I would also like to see an ack from @mhrivnak with a suggestion on the date this should be published and if he has any requested changes. Note that the date in the file description defines the date that Jekyll thinks the post was published on.

Thanks for putting this together! I can take another look to approve the changes when the next version is pushed.

File: _posts/2016-12-07-deprecating-nodes.md
Comment (mhrivnak) [inline _posts/2016-12-07-deprecating-nodes.md]: s/at //

File: _posts/2016-12-07-deprecating-nodes.md
Comment (mhrivnak) [inline _posts/2016-12-07-deprecating-nodes.md]: s/both servers are each a/each server is a/

File: _posts/2016-12-07-deprecating-nodes.md
Comment (mhrivnak) [inline _posts/2016-12-07-deprecating-nodes.md]: As a small detail: you'll see a relative path via this command, not a URL. I think it's helpful for us to be careful and consistent about calling something a URL vs. a path, to avoid ambiguity.

File: _posts/2016-12-07-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-07-deprecating-nodes.md]: s/at //

File: _posts/2016-12-07-deprecating-nodes.md
Comment (mhrivnak) [inline _posts/2016-12-07-deprecating-nodes.md]: I think in this sentence, it is a bit difficult to follow what it is saying about parent vs. child. I would eliminate most of what it says, and instead say those things below with the description of each cron entry.

"To make content available, you will need to call publish on the parent repository. This can be done automatically after each sync via the "auto-publish" feature, or periodically as shown here:"

File: _posts/2016-12-07-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-07-deprecating-nodes.md]: I know I wrote this sentence but...    s/even though it provides a similar feature set//

File: _posts/2016-12-07-deprecating-nodes.md
Comment (mhrivnak) [inline _posts/2016-12-07-deprecating-nodes.md]: Why? It would be valuable to explain this.

File: _posts/2016-12-07-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-07-deprecating-nodes.md]: s/any //

File: _posts/2016-12-07-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-07-deprecating-nodes.md]: s/call //
s/while/, and /

**Review by mhrivnak — CHANGES_REQUESTED** [review]
I made several suggestions.

I think it's also worth mentioning how to stay logged in. By default, a login only lasts 7 days. That's configurable in server.conf.

File: _posts/2016-12-07-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-07-deprecating-nodes.md]: Somehow can this be reworder for clarity. What I had meant was that pulp-admin commands destined for the child should be configured in cron on the child, and pulp-admin commands destined for the parent should be configured in cron on the parent. I'm not sure how to clarify this.

Comment (daviddavis) [reply _posts/2016-12-07-deprecating-nodes.md]: I see. How about something like:

> It's worth noting that you'll want to ensure that pulp-admin is pointing to the correct server. One option would be to edit the crontab for each server and have pulp-admin configured to point to Pulp via localhost.

File: _posts/2016-12-07-deprecating-nodes.md
Comment (bmbouter) [inline _posts/2016-12-07-deprecating-nodes.md]: s/ of/,/

File: _posts/2016-12-07-deprecating-nodes.md
Comment (bmbouter) [reply _posts/2016-12-07-deprecating-nodes.md]: @daviddavis that sounds great

**Review by bmbouter — APPROVED** [review]
Thank you so much @daviddavis . This looks ready to merge.

**Comment by mhrivnak** [comment]
Regarding my comment on login, was that addressed in any way? I haven't re-read the whole thing, but I don't initially see anything about login.

**Comment by daviddavis** [comment]
@mhrivnak apologies. I totally overlooked your comment. I added some info about login to the end of the blog post in the automatic sync section. Let me know if I need to tweak it. Thanks.

**Comment by bmbouter** [comment]
@daviddavis If this gets merged on Dec 8th please git mv the date once again. Once @mhrivnak signs off on it feel free to merge. Thanks!

**Review by mhrivnak — APPROVED** [review]

**Comment by daviddavis** [comment]
Merged. Thanks @bmbouter and @mhrivnak!

**Pull request merged**
